.github/workflows/pr-e2e.yml

```diff
@@ -89,13 +92,37 @@ jobs:
     steps:
       - uses: actions/checkout@v3
 
-      - name: Remove '${{ env.E2E_LABEL}}' label
-        id: remove-label
+      - name: Remove '${{ env.E2E_LABELS_NOTRUN}}' label
+        id: remove-not-run-label
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          git config --global --add safe.directory "$GITHUB_WORKSPACE"
+          gh pr edit ${{ needs.triage.outputs.pr_num }} --remove-label '${{ env.E2E_LABELS_NOTRUN}}'
+
+      - name: Remove '${{ env.E2E_LABELS_PASSED}}' label
+        id: remove-pass-label
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          git config --global --add safe.directory "$GITHUB_WORKSPACE"
+          gh pr edit ${{ needs.triage.outputs.pr_num }} --remove-label '${{ env.E2E_LABELS_PASSED}}'
+
+      - name: Remove '${{ env.E2E_LABELS_FAILED }}' label
+        id: remove-failed-label
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: |
           git config --global --add safe.directory "$GITHUB_WORKSPACE"
-          gh pr edit ${{ needs.triage.outputs.pr_num }} --remove-label '${{ env.E2E_LABEL}}'
+          gh pr edit ${{ needs.triage.outputs.pr_num }} --remove-label '${{ env.E2E_LABELS_FAILED }}'
+
+      - name: Add '${{ env.E2E_LABELS_RUNNING }}' label
+        id: add-running-label
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          git config --global --add safe.directory "$GITHUB_WORKSPACE"
+          gh pr edit ${{ needs.triage.outputs.pr_num }} --add-label '${{ env.E2E_LABELS_RUNNING }}'
```

Comment:
Should we merge these 3 steps into a single one? We can reduce a lot of lines just appending the `gh` lines to one of them

Reply:
Good idea